src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -702,6 +708,14 @@ C2V_VMENTRY_NULL(jobject, lookupJClass, (JNIEnv* env, jobject, jlong jclass_valu
     return JVMCIENV->get_jobject(result);
 C2V_END
 
+C2V_VMENTRY_0(jlong, getJObjectValue, (JNIEnv* env, jobject, jobject constant_jobject))
+    requireNotInHotSpot("getJObjectValue", JVMCI_CHECK_0);
+    JVMCIObject constant = JVMCIENV->wrap(constant_jobject);
+    Handle constant_value = JVMCIENV->asConstant(constant, JVMCI_CHECK_0);
+    jobject jni_handle = JNIHandles::make_local(THREAD, constant_value());
```

Comment:
Who is responsible for cleaning up these local refs?  My recollection is that they are allocated in the top most containing scope but if you haven't introduced new JNI scope they will never get released.

Reply:
We reset JNIHandleBlock::_top to 0 when exiting the native method frame, so I don't think a local ref wrapped as a jlong is safe to return here, as it will be stale when the caller returns.  You probably need a global ref and manage the lifecycle somehow.

Reply:
Can this method return jobject instead?  I have similar concerns as @fisk.  I admit I don't really understand how the constant if created or used.

Reply:
@tkrodriguez The Truffle shared library compiler is always invoked from Java code by calling a native method. The JNI local reference is allocated within the JNI locals frame of the native method. When the method ends, this frame is removed, and the handle is cleared.

For host method compilation, where the caller thread lacks a Java frame anchor, a new JNI local frame is pushed, which is subsequently popped at the end of the method. In this case, the handle is cleared by the `PopLocalFrame`.

Reply:
The `getJObjectValue` Javadoc has beed improved to explain how the returned JNI handle is scoped.